### PR TITLE
Updates the example to match the API

### PR DESCRIPTION
**Pull request by dalaing**
Hopefully this is the right change.

I also needed to import some addition things and turn on OverloadedStrings.  Let me know if you want those things in the full application in the example and I'll take care of it.

### CODE REVIEW

**Comment by ocharles** [comment]
I suppose it doesn't hurt to have `OverloadedStrings` in the example, because that is the idiomatic way of writing that code. If you can update this pull request with all the changes that permit a copy-and-paste build to work, then we'll get this merged.

**Comment by nkpart** [comment]
Please add the imports. I thought that because `V4` and `unless` weren't imported, I was missing an import for KeyDown as well and went hunting for it.

**Comment by dalaing** [comment]
Cool - updated.  I haven't done much in the way of code blocks in haddocks, so let me know if this needs any other changes.

File: src/SDL.hs [hunk 1]
```diff
@@ -118,7 +118,12 @@ If q was not pressed, we loop again. Otherwise, we exit the loop:
 To recap, here is our full application
 
 @
+{&#45;\# LANGUAGE OverloadedStrings \#&#45;}
```
Comment (ocharles) [inline src/SDL.hs]: Is the escaping intentional here?

**Comment by ocharles** [comment]
Looks good! Just want to make sure that escaping was intentional, then I'll merge and update the docs on Hackage :)

**Comment by dalaing** [comment]
It was intentional, but based on some information from before a haddock bugfix that made it easier to document those pragmas :) Fixed now.

**Comment by ocharles** [comment]
Thanks!

**Pull request merged**